crates/services/gas_price_service/src/v1/da_source_service.rs

```diff
@@ -191,11 +197,12 @@ mod tests {
         let latest_l2_height = latest_l2_height(l2_height);
         let (sender, mut receiver) =
             tokio::sync::broadcast::channel(DA_BLOCK_COSTS_CHANNEL_SIZE);
+        let expected_height = BlockHeight::new(recorded_height);
         let mut service = DaSourceService::new_with_sender(
             da_block_costs_source,
             Some(Duration::from_millis(1)),
             latest_l2_height,
-            None,
+            expected_height,
```

Comment:
I'm not sure I understand this test. 
You start and run `DaSourceService` with a block_height of`expected_height`, but then after running the service the recorded_height is still the same? This seems to be the opposite of what the test name suggests.

Should it be 
```suggestion
            BlockHeight::new(2),
```
instead?

Reply:
perhaps.. theres a bunch of things that are wrong in this pr now after #2512 was merged.

Reply:
cool, I didn't see it was back in draft. Will stop reviewing until it's ready for review again

Reply:
addressed in 73cf35beef